### ISSUE TITLE
Objects in attributes are considered relationships 

### ISSUE THREAD

**Issue by raoufswe** [wopian/kitsu#397]
The library currently sees any object in attributes as a relationship, however, based on the [JSONAPI spec](https://codesandbox.io/s/magical-wescoff-f79zc?file=/src/App.js), objects and arrays are allowed as attribute values.

here is a demo of the issue [codesandbox](https://codesandbox.io/s/magical-wescoff-f79zc?file=/src/App.js)

**Comment by raoufswe** [comment]
there is a possibility that one of the API endpoints i will be working on has this edge case, so I'm happy to work in a PR if it seems a valid thing to add to the library.

**Comment by wopian** [comment]
Yeah, guessing the type was a bad idea. Enforcing `type` to exist and only turn an array/object into a relationship if it exists would be the solution for a 8.0.0 release.

I unfortunately never had to serialise arrays or objects for the API I use this with (although I do consume a lot of them in the deserialise stage) - so this problem eluded me and surprised it wasn't included in the test cases.

**Comment by wopian** [comment]
I believe this can be released without being a breaking change.

### Changes:
- Checks relationship object/array to see if it contains the `id` attribute **(new)**
	- `id` and `type` are required for relationship POST/PATCH requests in spec
	- `type` is either supplied by the user or guessed from the variable name (existing behaviour)
	- if `id` exists, object/array become a relationship (existing behaviour)
	- if `id` doesn't exist, object/array become an attribute **(new behaviour)**
- No longer errors if `id` is missing from relationship **(new)**
	- Impossible to differentiate between a regular object and a relationship object/array that's missing ID to throw this error
		- Would probably be a breaking change if `id` wasn't always required in the spec, so I don't feel it warrants a major version
- `null` and `[]` resolve to `data: null` and `data: []` to delete relationships instead of erroring **(new)**

### Example:
```js
serialise('resourceModel', {
	array: [ 0 ],
	array2: [ {
    	id: '2',
    	type: 'relationshipModel'
    } ],
	array3: {
   		id: '3'
    }
})
```

```json5
data: {
	type: 'resourceModel',
	attributes: {
		array: [ 0 ]
	},
	relationships: {
		array2: {
			data: [ {
				id: '2',
				type: 'relationshipModel'
			} ]
		},
		array3: {
			data: {
				id: '3',
				type: 'array3'
			}
		}
	}
}
```

**Comment by wopian** [comment]
Released 8.3.0
